supervisor/shared/external_flash/external_flash.c

```diff
@@ -191,41 +198,49 @@ void supervisor_flash_init(void) {
 
     spi_flash_init();
 
-    // The response will be 0xff if the flash needs more time to start up.
-    uint8_t jedec_id_response[3] = {0xff, 0xff, 0xff};
-    while (jedec_id_response[0] == 0xff) {
-        spi_flash_read_command(CMD_READ_JEDEC_ID, jedec_id_response, 3);
-    }
-
-    for (uint8_t i = 0; i < EXTERNAL_FLASH_DEVICE_COUNT; i++) {
-        const external_flash_device* possible_device = &possible_devices[i];
-        if (jedec_id_response[0] == possible_device->manufacturer_id &&
-            jedec_id_response[1] == possible_device->memory_type &&
-            jedec_id_response[2] == possible_device->capacity) {
+#ifdef EXTERNAL_FLASH_QSPI_SINGLE
+        // For NVM that don't have JEDEC response
+        spi_flash_command(CMD_WAKE);
+        for (uint8_t i = 0; i < EXTERNAL_FLASH_DEVICE_COUNT; i++) {
+            const external_flash_device* possible_device = &possible_devices[i];
```

Comment:
Again, particular to the chip, so something descriptive bit like `no_jedec_response` would be appropriate.